### PR TITLE
improve doc of std.math.sin

### PR DESCRIPTION
**Pull request by WalterBright**
Add detail, example, and helpful links.

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -566,14 +568,33 @@ unittest
...
- * Returns sine of x. x is in radians.
+ * Returns $(WEB en.wikipedia.org/wiki/Sine, sine) of x. x is in $(WEB en.wikipedia.org/wiki/Radian, radians).
  *
  *      $(TABLE_SV
- *      $(TR $(TH x)               $(TH sin(x))      $(TH invalid?))
- *      $(TR $(TD $(NAN))          $(TD $(NAN))      $(TD yes))
- *      $(TR $(TD $(PLUSMN)0.0)    $(TD $(PLUSMN)0.0) $(TD no))
- *      $(TR $(TD $(PLUSMNINF))    $(TD $(NAN))      $(TD yes))
+ *      $(TH3 x           ,  sin(x)      ,  invalid?)
+ *      $(TD3 $(NAN)      ,  $(NAN)      ,  yes     )
```
Comment (yebblies) [inline std/math.d]: Did you mean TR3?

Comment (WalterBright) [inline std/math.d]: No, made the typo further up. Good catch.

**Comment by quickfur** [comment]
Is there a good reason the code example isn't a ddoc'd unittest? That ensures it's actually runnable, and actually does what it claims to.

**Comment by WalterBright** [comment]
done

File: std/math.d [hunk 2]
```diff
@@ -566,23 +568,43 @@ unittest
...
+    import std.math : sin, PI;
+    import std.stdio : writefln;
+
+    void main()
+    {
+      real x = 30.0;
+      auto result = sin(x * (PI / 180)); // convert degrees to radians
+      writefln("The sine of %s degrees is %s", x, result);
+    }
+}
```
Comment (quickfur) [inline std/math.d]: Uhh? This unittest does nothing at all (it just declares a function). How is this supposed to be any different from an untested code example?

Comment (WalterBright) [inline std/math.d]: It verifies that it compiles.

The trouble with testing the result of transcendental functions is they are inexact and have to be tested with "fuzzy" tests. This is not appropriate for example/tutorial code.

**Pull request merged**